src/components/Filters/StatefulFilters.tsx

```diff
@@ -75,17 +75,7 @@ export class StatefulFilters extends React.Component<StatefulFiltersProps, State
       }
     });
 
-    this.filterTypePromise = makeCancelablePromise(Promise.all(promises));
-    this.filterTypePromise.promise
-      .then(types => {
-        this.setState({ filterTypes: types });
-        this.filterTypePromise = undefined;
-      })
-      .catch(err => {
-        if (!err.isCancelable) {
-          console.debug(err);
-        }
-      });
+    this.promises.registerAll('filterType', promises).then(types => this.setState({ filterTypes: types }));
```

Comment:
Having a local const promises with a global this.promises makes a little bit harder to read.
I suggest to change one of them to avoid confusion.